modules/task/backend.go

```diff
@@ -38,14 +38,20 @@ const (
 type Backend interface {
 	service.Module
 	Encoder() Encoding
-	Publish(ctx context.Context, message []byte) error
+	Enqueue(ctx context.Context, message []byte) error
+	ExecuteAsync() chan error
```

Comment:
Could you add some comments, what does this function supposed to do and what kind of channel does it return?